### PR TITLE
Add Abseil as a dep and update RE2

### PR DESCRIPTION
**Pull request by s-perron**
The latest version of RE2 requires Abseil. This PR adds Abseil as an
external dependence, and update RE2 to use it.

### CODE REVIEW

File: external/CMakeLists.txt [hunk 1]
```diff
@@ -97,6 +97,21 @@ if (NOT ${SPIRV_SKIP_TESTS})
...
+
+    if(NOT TARGET absl::base)
+      set(ABSL_INTERNAL_AT_LEAST_CXX17 ON)
+      set(ABSL_PROPAGATE_CXX_STD ON)
+      set(ABSL_ENABLE_INSTALL ON)
+      add_subdirectory(abseil_cpp EXCLUDE_FROM_ALL)
+    endif()
+
+    if(NOT TARGET absl_base)
+      message(STATUS "STEVEN: abseil is not a valid target")
```
Comment (dnovillo) [inline external/CMakeLists.txt:109]: debug code leftover?

**Review by dnovillo — CHANGES_REQUESTED** [review]
more dependencies!  what fun...

File: external/CMakeLists.txt (hunk 1 above)
Comment (s-perron) [reply external/CMakeLists.txt]: :man_facepalming:

**Review by dnovillo — APPROVED** [review]

**Comment by LukasBanana** [comment]
@s-perron @dnovillo since I'm running into build issues due to this new dependency, I was wondering why we can't simply add all these dependencies as submodules to simplify git clone process? Or am I missing something from the build documentation?

**Comment by dnovillo** [comment]
@LukasBanana I sometimes wish SPIRV-Tools used submodules, but it uses the "sync deps scripts" model (https://github.com/KhronosGroup/SPIRV-Tools#getting-the-source).  I'm not actually sure why abseil is not updated that way, though.  @s-perron maybe we should fix CMake to use `abseil-cpp`?

**Comment by LukasBanana** [comment]
I just noticed that _effcee_ has this CMake condition:
```
  if (IS_DIRECTORY "${EFFCEE_THIRD_PARTY_ROOT_DIR}/abseil_cpp")
    set(EFFCEE_ABSEIL_DIR "${EFFCEE_THIRD_PARTY_ROOT_DIR}/abseil_cpp" CACHE STRING "Location of abseil source")
  elseif(IS_DIRECTORY "${EFFCEE_THIRD_PARTY_ROOT_DIR}/abseil-cpp")
    set(EFFCEE_ABSEIL_DIR "${EFFCEE_THIRD_PARTY_ROOT_DIR}/abseil-cpp" CACHE STRING "Location of abseil source")
  endif()
```
But I'm not sure how that ties in with the SPIRV-Tools CMake script.

**Pull request merged**